### PR TITLE
allow re-enabling disabled sources

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3843

### CODE REVIEW

File: suricata/update/sources.py [hunk 1]
```diff
@@ -44,6 +44,15 @@ def get_source_directory():
 def get_index_filename():
     return os.path.join(config.get_cache_dir(), SOURCE_INDEX_FILENAME)
 
+def get_sources_from_dir():
+    """Return names of all files existing in the sources dir"""
+    source_dir = get_source_directory()
+    source_names = []
+    # TODO: Check if this will work with Python 3
```
Comment (inashivb) [inline suricata/update/sources.py:51]: Oops. Meant Python 2 here.

**Comment by inashivb** [comment]
Note to self: Fix spelling in commit message.

**Comment by inashivb** [comment]
@jasonish could you please see if this is the right fix for the linked issue? Other way was to allow re enabling the issue on add-source but I think it goes against how you probably intended it to work (it throws an error in that case).
This looked like an easy fix.

Please let me know if it misses some important use case that I might be missing,

**Comment by jasonish** [comment]
This seems to work. I don't see the index being involved here, as the reporter was using a self-defined ruleset that is not part of the index.

However, the reported bug appears to only be in the display only..  For example, with master:
<pre>
./bin/suricata-update add-source datasets https://codemonkey.net/datasets.tar.gz
2/9/2020 -- 16:47:49 - <Info> -- Using data-directory /var/lib/suricata.
2/9/2020 -- 16:47:49 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
2/9/2020 -- 16:47:49 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
2/9/2020 -- 16:47:49 - <Info> -- Found Suricata version 6.0.0-dev at /usr/bin/suricata.
2/9/2020 -- 16:47:49 - <Info> -- Creating directory /var/lib/suricata/update/sources
</pre>

<pre>
./bin/suricata-update disable-source datasets
2/9/2020 -- 16:48:27 - <Info> -- Using data-directory /var/lib/suricata.
2/9/2020 -- 16:48:27 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
2/9/2020 -- 16:48:27 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
2/9/2020 -- 16:48:27 - <Info> -- Found Suricata version 6.0.0-dev at /usr/bin/suricata.
2/9/2020 -- 16:48:27 - <Info> -- Source datasets has been disabled
</pre>

<pre>
./bin/suricata-update list-enabled-sources
2/9/2020 -- 16:49:29 - <Info> -- Using data-directory /var/lib/suricata.
2/9/2020 -- 16:49:29 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
2/9/2020 -- 16:49:29 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
2/9/2020 -- 16:49:29 - <Info> -- Found Suricata version 6.0.0-dev at /usr/bin/suricata.
2/9/2020 -- 16:49:29 - <Warning> -- No enabled sources.
</pre>

<pre>
./bin/suricata-update enable-source datasets
2/9/2020 -- 16:49:47 - <Info> -- Using data-directory /var/lib/suricata.
2/9/2020 -- 16:49:47 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
2/9/2020 -- 16:49:47 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
2/9/2020 -- 16:49:47 - <Info> -- Found Suricata version 6.0.0-dev at /usr/bin/suricata.
2/9/2020 -- 16:49:47 - <Info> -- Re-enabling previously disabled source for datasets.
2/9/2020 -- 16:49:47 - <Error> -- Unknown source: datasets
</pre>

<pre>
./bin/suricata-update list-enabled-sources
2/9/2020 -- 16:50:01 - <Info> -- Using data-directory /var/lib/suricata.
2/9/2020 -- 16:50:01 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
2/9/2020 -- 16:50:01 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
2/9/2020 -- 16:50:01 - <Info> -- Found Suricata version 6.0.0-dev at /usr/bin/suricata.
Enabled sources:
  - datasets
</pre>

So even though it error'd out, the `datasets` source did get re-enabled.

**Review by jasonish — APPROVED** [review]
I think this is good.

**Comment by inashivb** [comment]
Merged through https://github.com/OISF/suricata-update/pull/252
Thank you!

**Pull request closed without merging**